pkg/machines/libvirt-common.es6

```diff
@@ -122,6 +122,29 @@ function getBootableDeviceType(device) {
     return type;
 }
 
+export function getDiskElemByTarget(domxml, targetOriginal) {
+    const domainElem = getDomainElem(domxml);
+
+    if (!domainElem) {
+        console.warn(`Can't parse dumpxml, input: "${domainElem}"`);
+        return;
+    }
+
+    const devicesElem = domainElem.getElementsByTagName("devices")[0];
+    const diskElems = devicesElem.getElementsByTagName('disk');
+
+    if (diskElems) {
+        for (let i = 0; i < diskElems.length; i++) {
+            const diskElem = diskElems[i];
```

Comment:
Iteration like this is fine, if you prefer this. But just in case you didn't know: It's shorter, and more robust (avoids off-by-one errors and suchlike) to do

    (diskElems || []).map(diskElem => { ...})

In this case you want to stop on the first hit, so you could use  instead use [find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead of `.map()`.

Reply:
Done